sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -212,6 +212,12 @@ class PrefixKeyScanStateEncoder(
  * We cannot support variable sized fields given the UnsafeRow format which stores variable
  * sized fields as offset and length pointers to the actual values, thereby changing the required
  * ordering.
+ * Note that we also support "null" values being passed for these fixed size fields. We prepend
+ * a single byte to indicate whether the column value is null or not. We cannot change the
+ * nullability on the UnsafeRow itself as the expected ordering would change if non-first
+ * columns are marked as null. If the first col is null, those entries will appear last in
+ * the iterator. If non-first columns are null, ordering based on the previous columns will
```

Comment:
This comment actually confused me. If I'm not missing here, it's just applying "null-last" for each column, and ordering should just work as the same. E.g. `(1, 1, 3, 1, 2), (1, 1, null, 1, 3)`, `(1, 1, null, 2, 2)` should be still ordered. Having null in the middle of column does not block the data to be ordered for remaining columns.

Please correct me if I'm missing something. If I'm on the right track, could we please update the comment and also add a test to check this behavior? Probably adding more values in the latest test case would work.

Reply:
Not exactly sure what you mean -

for eg - if you look at the test here - https://github.com/apache/spark/pull/45503/files#diff-4c6b19c847c68e25ffa927df85efb4c79f388648d8c6242f1fe9f84cf09ec5ffR449

Suppose we have the sample input as:
```
Seq((931L, 10), (40L, null), (452300L, 1), (1L, 304), (100L, null))
```

when we iterate the long elements should all be ordered as expected

So we should get
```
Seq(1L, 40L, 100L, 931L, 452300L)
```

So in this case the ordering on the first col will still apply which is what I was trying to convey

Reply:
What I meant is, the columns after null values should be still ordered. For the test, we should still be able to iterate the data of (null, 1), (null, 3), (null, 2) as ordered, (null, 1), (null, 2), (null, 3).

Reply:
Done- added a new test, expanded the previous test and also updated the comment a bit.